Tahi-project/tahi
# Feature/aperta 9622 required for submission

JIRA issue: https://jira.plos.org/jira/browse/APERTA-9622

#### What this PR does:

Adds a `required-for-submission` flag to card versions.

WIth this flag, a custom card will be required for submission if the flag is set.

Contrary to my pervious assertions, this can be POed and tested before the permissions stuff are done, if you are logged in as admin.

To test:
1. Login as an admin
2. Create a paper in the "Yeti" journal with the "All custom" type (scroll down to see it)
3. Done!
 
![submit](https://cloud.githubusercontent.com/assets/22718/24781427/85c6ad5e-1af4-11e7-889a-c0fe9b8cd0ba.gif)

---

#### Code Review Tasks:

Author tasks:

- [X] If I changed the database schema, I enforced database constraints.

Reviewer tasks:


#### After the Code Review:

Author tasks:
